expan/core/util.py

```diff
@@ -1,10 +1,20 @@
+import json
+from enum import Enum
 from warnings import warn
 
 import numpy as np
 import pandas as pd
 from datetime import datetime, timedelta
 
 
+class JsonSerializable(object):
+    def toJson(self):
+        return json.dumps(self, default=lambda o: o.name if isinstance(o, Enum) else o.__dict__, sort_keys=True, indent=4)
```

Comment:
Would this work too for nested `JsonSerializable`, i.e. if one instance of `JsonSerializable` has another instance of `JsonSerializable` as one of its fields?

Reply:
yes it works. you can run ```pytest -s test_json_serializer.py``` to see the output of those nested result classes.